e3sm_diags/parser/core_parser.py

```diff
@@ -477,11 +568,631 @@ def load_default_args(self, files=[]):
             required=False,
         )
 
-        self.add_argument(
+        self.parser.add_argument(
             "--debug",
             dest="debug",
             help="Turns debugging on, allows code to prematurely break.",
             action="store_const",
             const=True,
             required=False,
         )
+
+    def parse_args(
+        self,
+        args: Optional[List[str]] = None,
+        namespace: Optional[argparse.Namespace] = None,
+    ) -> argparse.Namespace:
+        """Parses arguments passed from the command line.
+
+        This method records the command used to run the parser, which is either
+        `args` or `sys.argv`.
+
+        Parameters
+        ----------
+        args : List[str], optional
+            The list of arguments, by default [].
+        namespace : Optional[argparse.Namespace], optional
+            The namespace, by default None
+
+        Returns
+        -------
+        Namespace
+            The argparse.Namespace object with the parsed arguments.
+        """
+        # Remove arguments set by `ipykernel` via `sys.argv` because they
+        # are not defined and recognized by `self.parser` using `add_argument()`.
+        sys.argv = self._remove_ipykernel_args()
+        self.cmd_used = sys.argv if args is None else args
+
+        namespace, _ = self.parser.parse_known_args(args, namespace)
+
+        return namespace
```

Comment:
I figured out that tools like `ipykernel` and `pytest` add arguments to `sys.argv`, which breaks `argparse.ArgumentParser`.

I created a function that removes `ipykernel` arguments from `sys.argv`. We can now run `e3sm_diags` interactively using any `ipykernel` based tooling (e.g., Jupyter Notebook, VSCode's debugger). This is really useful for stepping through the code as we refactor the diagnostic sets.

Reply:
This is great improvement resulting from this work!